src/phpDocumentor/ConfigurationFactory/PhpDocumentor3.php

```diff
@@ -0,0 +1,176 @@
+                    'output' => $outputDirectory,
+                    'cache'  => $cacheDirectory,
+                ],
+                'versions'  => $versions,
+                'templates' => $template,
+            ],
+        ];
+
+        return $phpdoc3Array;
+    }
+
+    /**
+     * Checks if the provided configuration xml is for phpDocumentor3.
+     *
+     * @param $phpDocumentor
+     *
+     * @return bool
+     */
+    public function match(\SimpleXMLElement $phpDocumentor)
+    {
+        return (string) $phpDocumentor->attributes()->version === '3';
+    }
+
+    /**
+     * Builds the versions part of the array from the phpDocumentor3 configuration xml.
+     *
+     * @param \SimpleXMLElement $version
+     *
+     * @return array
+     */
+    private function buildVersions(\SimpleXMLElement $version)
+    {
+        $extensions = [];
+        foreach ($version->api->extensions->children() as $extension) {
+            if ((string) $extension !== '') {
+                $extensions[] = (string) $extension;
+            }
+        }
+
+        $ignoreHidden   = filter_var($version->api->ignore->attributes()->hidden, FILTER_VALIDATE_BOOLEAN);
+        $ignoreSymlinks = filter_var($version->api->ignore->attributes()->symlinks, FILTER_VALIDATE_BOOLEAN);
+
+        return [
+            'folder' => (string) $version->folder,
+            'api'    => [
+                'format'               => ((string) $version->api->attributes()->format) ?: 'php',
+                'source'               => [
+                    'dsn'   => (string) $version->api->source->attributes()->dsn,
+                    'paths' => ((array) $version->api->source->path) ?: ['src'],
+                ],
+                'ignore'               => [
+                    'hidden'   => $ignoreHidden,
+                    'symlinks' => $ignoreSymlinks,
+                    'paths'    => ((array) $version->api->ignore->path) ?: [],
+                ],
+                'extensions'           => $extensions,
+                'visibility'           => (string) $version->api->visibility,
+                'default-package-name' => ((string) $version->api->{'default-package-name'}) ?: 'Default',
+                'markers'              => (array) $version->api->markers->children()->marker,
+            ],
+            'guide'  => [
+                'format' => ((string) $version->guide->attributes()->format) ?: 'rst',
+                'source' => [
+                    'dsn'   => ((string) $version->guide->source->attributes()->dsn) ?: 'file://.',
+                    'paths' => (array) $version->guide->source->path,
+                ],
+            ],
+        ];
+    }
+
+    /**
+     * Builds the template part of the array from the phpDocumentor3 configuration xml.
+     *
+     * @param \SimpleXMLElement $template
+     *
+     * @return array
+     */
+    private function buildTemplate(\SimpleXMLElement $template)
+    {
+        if (!$template) {
```

Comment:
could you make this check more explicit? $template will never be false, I can imagine that there are situations that are not valid. But I would like to see what situation is is. Php will convert this situation to false. but that is not what it is. 

$template === false will never be true.